util/chunk/chunk.go

```diff
@@ -277,6 +278,60 @@ func (c *Chunk) AppendPartialRow(colIdx int, row Row) {
 	}
 }
 
+// PreAlloc4Row pre-allocates the memory space for a Row.
+// The null elem info will be pre-written.
+func (c *Chunk) PreAlloc4Row(row Row) {
```

Comment:
Can we preallocate the memory for a batch of rows? The memory grow stratagem may allocate a lot of unused memory.

Reply:
1. Preallocate memory for a batch of rows cannot avoid the waste problem brought by memory usage increment of chunk.data.
Unless the batch contains **all** of the rows which should be pre-allocated.


2. I adjust the strategy according to https://github.com/golang/go/blob/master/src/runtime/slice.go#L116-L135